File: commons-mis/src/main/java/io/github/cooperlyt/mis/work/impl/model/WorkTaskModel.java
package io.github.cooperlyt.mis.work.impl.model;


import io.github.cooperlyt.mis.work.data.TaskMessage;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Table("work_task")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkTaskModel implements TaskMessage, Persistable<String> {

  @Id
  @Column("task_id")
  private String id;

  private String message;

  private String taskName;

  private Boolean pass;

  @Override
  public boolean isNew() {
    return true;
  }
}

File: commons-mis/src/main/java/io/github/cooperlyt/mis/work/impl/model/WorkTaskModel.kt
package io.github.cooperlyt.mis.work.impl.model

import io.github.cooperlyt.mis.work.data.WorkTask
import org.springframework.data.annotation.Id
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table

@Table("work_task")
data class WorkTaskModel(
  @Id
  @Column("task_id")
  override val taskId: String,
  override val taskName: String,
  override val message: String?,
  override val pass: Boolean
): WorkTask, Persistable<String> {
  override fun getId(): String? = taskId

  override fun isNew(): Boolean = true
}
